pmlc/dialect/tile/ir/ops.cc

```diff
@@ -719,12 +719,20 @@ void ReshapeOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
 
 Type ReshapeOp::getResultType(ArrayRef<Value> operands) {
   IVLOG(5, "ReshapeOp::getResultType>")
-  if (operands.size() < 2) {
-    throw std::runtime_error("ReshapeOp requires at least 2 operands");
+  if (!operands.size()) {
+    throw std::runtime_error("ReshapeOp requires at least 1 operand");
   }
   auto tensor = operands.front();
-  auto dims = operands.drop_front();
   auto tensorType = eltwise::getRankedTensorType(tensor.getType());
+  // In the case of scalars, return the original type
+  if (!tensorType.getRank()) {
+    return tensorType;
+  }
+  auto dims = operands.drop_front();
+  if (!dims.size()) {
+    throw std::runtime_error(
+        "ReshapeOp with Tensor of rank > 0 requires dims to be set");
+  }
   auto elementType = tensorType.getElementType();
```

Comment:
I don't think this is exactly right, imagine if the user wanted to reshape a `tensor<1xf32>` into a `tensor<f32>`, that should be legal shouldn't it? What we could do though is check if both the source and target are both rank 0, then return the operand. However, we should have a test for the N-rank to 0-rank reshape when all the dims of the N-rank tensor are 1.

Reply:
Likewise, we should be able to reshape a 0-rank tensor into an N-rank tensor, provided the `dims` are all 1s.

Reply:
That makes sense. I wasn't thinking of the N-rank to 0-rank reshape case when I wrote this. I agree that `ComputeShape` should handle this.